packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -913,7 +938,7 @@ export class ContainerRuntime
 			chunkSizeInBytes = defaultChunkSizeInBytes,
 			enableGroupedBatching = true,
 			explicitSchemaControl = false,
-		} = runtimeOptions;
+		} = runtimeOptions as IContainerRuntimeOptionsInternal;
```

Comment:
We really should avoid `as` it can do some questionable things. A little silly but you can instead assign to local variable of the right type first to avoid `as`.
```typescript
		const runtimeOptionsInternal: IContainerRuntimeOptionsInternal = runtimeOptions;
```
Why does it matter? Well, if you added a required property on `IContainerRuntimeOptionsInternal` then `as` cast won't care but the assignment would properly fail.

Reply:
Interesting.  When I was reviewing again my use of `as` in this PR, this one felt pretty good because I'm immediately destructuring - it's just letting me find extra stuff in there.  So I think it's ok?

Reply:
i must have missed this, when i re-reviewd. you can just do:
```
		const {
			summaryOptions = {},
			gcOptions = {},
			loadSequenceNumberVerification = "close",
			flushMode = defaultFlushMode,
			compressionOptions = defaultCompressionConfig,
			maxBatchSizeInBytes = defaultMaxBatchSizeInBytes,
			enableRuntimeIdCompressor,
			chunkSizeInBytes = defaultChunkSizeInBytes,
			enableGroupedBatching = true,
			explicitSchemaControl = false,
		} = runtimeOptions satisfies IContainerRuntimeOptionsInternal;
```
`as` is always bad, sometimes necessary but always bad, as we should always try to avoid it

Reply:
It is not really okay if you want verification at compile time. There are far too many uses of `as` in the code. Good news is that I tried an alternate pattern and it works as desired. Use:
```typescript
}: IContainerRuntimeOptionsInternal = runtimeOptions;
```

Reply:
Nice, taking this suggestion in #23309